commands/aliases.go

```diff
@@ -0,0 +1,203 @@
+		if err := ioutil.WriteFile(aliasStorageFilepath, []byte("{}"), 0744); err != nil {
+			return fmt.Errorf("Failed to create alias file: %s", err)
+		}
+	}
+
+	file, err := ioutil.ReadFile(aliasStorageFilepath)
+	if err != nil {
+		return fmt.Errorf("failed to read aliases file: %s", err)
+	}
+
+	if err := json.Unmarshal(file, &aliasMap); err != nil {
+		return fmt.Errorf("failed to unmarshal alias JSON: %s", err)
+	}
+
+	return nil
+}
+
+// writeToStorage overwrites the aliases held within `aliasMap` to the storage file.
+// Note that if an alias was removed from `aliasMap`, the removed alias will also
+// be removed from the storage file.
+func writeToStorage() error {
+	aliasBytes, err := json.Marshal(aliasMap)
+	if err != nil {
+		return fmt.Errorf("Failed to marshal JSON: %s", err)
+	}
+	if err := ioutil.WriteFile(aliasStorageFilepath, aliasBytes, 0744); err != nil {
+		return fmt.Errorf("Failed to write alias file: %s", err)
+	}
+	return nil
+}
+
+// withAliasCommands takes a map of command names to the command itself and
+// adds in a command where the command name is the alias key and the command
+// itself is a command which returns the alias value.
+func withAliasCommands(commMap map[string]Command) (map[string]Command, error) {
+	if err := loadFromStorage(); err != nil {
+		return nil, fmt.Errorf("failed to load aliases from storage: %s", err)
+	}
+
+	for aliasName, aliasValue := range aliasMap {
+		commMap[aliasName] = &textCommand{
+			helpText: aliasValue,
+			command: func(_ context.Context, _ []string) (string, error) {
+				return aliasValue, nil
+			},
+		}
+	}
+	return commMap, nil
+}
+
+// aliasCommand sets string => string shortcut that can be called later to print a value
+func aliasCommand(ctx context.Context, args []string) (string, error) {
+	l, loggerOk := logging.FromContext(ctx)
+	if loggerOk {
+		l.Infof("Responding to alias command")
+	}
+
+	if len(args) == 2 {
+		return "", errors.New("Too few arguments supplied. Refer to !help for usage")
+	}
+	if len(args) == 1 {
+		// shop a paginator of all aliases
+		p := dgwidgets.NewPaginator(
+			ctx.Value(messageContextValue("Session")).(*discordgo.Session),
+			ctx.Value(messageContextValue("ChannelID")).(string))
+		p.Add(&discordgo.MessageEmbed{
+			Color: 0,
+
+			Fields: []*discordgo.MessageEmbedField{
+				{
+					Name: "Aliases",
+					Value: func() string {
+						var sortedAliases []string
+						for a := range aliasMap {
+							sortedAliases = append(sortedAliases, fmt.Sprintf("**%s**", a))
+						}
+						sort.Strings(sortedAliases)
+						for i := range sortedAliases {
+							sortedAliases[i] = fmt.Sprintf("%d) %s", i+2, sortedAliases[i])
+						}
+						return strings.Join(sortedAliases, "\n")
+					}(),
+				},
+			},
+		})
+
+		for aliasName, aliasValue := range aliasMap {
+			embed := &discordgo.MessageEmbed{
+				Title:       aliasName,
+				Description: aliasValue,
+			}
+
+			resp, err := http.Head(aliasValue)
```

Comment:
Evan made a point that we could pre-compute this, as doing it like this makes the listing slow as we're checking everything. Maybe you wanna do this? Else we can do it at a later stage

Reply:
That's a good point. It's a change that probably doesn't fit in this PR. It will require changing the way aliases are stored. Not going to do it now but will do it in a different PR.

Reply:
Agreed :ok_hand: